src/client/services/feature/feature.js

```diff
@@ -61,7 +62,7 @@ var FeatureService = SparkBase.extend(
     return this.request({
       method: 'POST',
       api: 'feature',
-      resource: 'features/users/' + this.spark.device.userId + '/developer',
```

Comment:
@ianwremmel  The test were passing because after the api request was made .. The then block would always set the correct value in the feature store .  
here it was 'developer' 
 resource: 'features/users/' + this.spark.device.userId + '/developer,

But after then we were using correct value so the test was always passing
return this.spark.device.features[keyType].add(res.body, {merge: true});

I have added a check fro network request